tests/unit/test_checkPot/__init__.py

```diff
@@ -0,0 +1,90 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2020 NV Access Limited, Julien Cochuyt
+# This file may be used under the terms of the GNU General Public License, version 2 or later.
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+
+"""Unit tests for the checkPot test."""
+
+from contextlib import redirect_stdout, redirect_stderr
+from io import StringIO
+import os.path
+import unittest
+from typing import Tuple
+
+
+class TestCheckPot(unittest.TestCase):
+	"""Tests that checkPot works as expected."""
+
+	def setUp(self):
+		from ... import checkPot
+		self.checkPot = checkPot
+		self.saved_EXPECTED_MESSAGES_WITHOUT_COMMENTS = checkPot.EXPECTED_MESSAGES_WITHOUT_COMMENTS
+		checkPot.EXPECTED_MESSAGES_WITHOUT_COMMENTS = {
+			"Expected message without comment #1",
+			"Expected message without comment #2",
+		}
+
+	def tearDown(self):
+		self.checkPot.EXPECTED_MESSAGES_WITHOUT_COMMENTS = self.saved_EXPECTED_MESSAGES_WITHOUT_COMMENTS
+
+	def doCheckPot(self, fileName: str) -> Tuple[int, str]:
+		"""Run checkPot against the specified test POT file and returns its error count and status line."""
+		filePath = os.path.join(os.path.dirname(__file__), fileName)
+		bufOut = StringIO()
+		bufErr = StringIO()
+		with redirect_stdout(bufOut), redirect_stderr(bufErr):
+			errorCount = self.checkPot.checkPot(filePath)
+		self.assertTrue(bufOut.tell(), "No standard output")
+		self.assertFalse(bufErr.tell(), "Non-empty standard error output")
+		outLines = bufOut.getvalue().split("\n")
+		self.assertGreaterEqual(len(outLines), 2, "Less than two lines of standard output")
+		self.assertFalse(outLines[-1], "Standard output does not end with an empty line.")
+		statusLine = outLines[-2]
+		return errorCount, statusLine
+
+	def test_checkPot_firstMessage(self):
+		"""Test that missing translators comment on the first message are reported."""
+		self.assertEqual(
+			self.doCheckPot("firstMessage.pot"),
+			(1, "1 errors, 0 unexpected successes, 0 expected errors"),
+			"checkPot error count and/or status message do not meet expectations."
+		)
+
+	def test_checkPot_lastMessage(self):
+		"""Test that missing translators comment on the last message are reported."""
+		self.assertEqual(
+			self.doCheckPot("firstMessage.pot"),
```

Comment:
I think these tests would be easier to read if the test data was inline, preferably using a template so that just the test specific data is visible. On the other hand it's probably a good thing that this tests it end to end.

Reply:
Not easy to do without refactoring `checkPot` itself, as it currently only takes a filename parameter.
Furthermore, the test data currently sums up to 208 lines, which IMHO would reduce a whole lot the readability of the 98 lines long code.
Please note that the header of each data file also describes what errors are expected to be found in it.

Reply:
I agree for now this is fine. Mostly I'm thinking about the ease of reading the tests, going back and forth to data files to understand the cases being tested can be a pain. One benefit of succinct unit tests is the way they document code to developers, and invite developers to add variations to understand how the code works under different circumstances.